e2e/suite_test.go

```diff
@@ -0,0 +1,181 @@
+package e2e_test
+
+import (
+	"context"
+	"fmt"
+	mrand "math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Azure/agentbaker/pkg/agent"
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/barkimedes/go-deepcopy"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var cases = map[string]scenarioConfig{
+	"base": {},
+	// "gpu": {
+	// 	bootstrapConfigMutator: func(t *testing.T, nbc *datamodel.NodeBootstrappingConfiguration) {
+	// 		nbc.ContainerService.Properties.AgentPoolProfiles[0].VMSize = "Standard_NC6"
+	// 		nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-ubuntu-containerd-18.04-gen2"
+	// 		nbc.AgentPoolProfile.VMSize = "Standard_NC6"
+	// 		nbc.AgentPoolProfile.Distro = "aks-ubuntu-containerd-18.04-gen2"
+	// 		nbc.ConfigGPUDriverIfNeeded = true
+	// 		nbc.EnableGPUDevicePluginIfNeeded = false
+	// 		nbc.EnableNvidia = true
+	// 	},
+	// },
+}
+
+func Test_All(t *testing.T) {
+	r := mrand.New(mrand.NewSource(time.Now().UnixNano()))
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	suiteConfig, err := newSuiteConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cloud, err := newAzureClient(suiteConfig.subscription)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kube, err := getClusterKubeClient(ctx, cloud, suiteConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clusterParams, err := extractClusterParameters(ctx, t, kube)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	baseConfig, err := getBaseBootstrappingConfig(ctx, t, cloud, suiteConfig, clusterParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for name, tc := range cases {
```

Comment:
Looks like this is done serially? Can we parallelize the test case runs?

Reply:
t.Run actually already runs parallel in this setup I believe

Reply:
added t.Parallel to make it work